Assets/Scripts/NPC/Waypoint.cs

```diff
@@ -0,0 +1,73 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public sealed class Waypoint : MonoBehaviour
+{
+    [Header("Grid")]
+    [SerializeField] private Grid parentGrid;
+    [SerializeField] private bool isOccupied;
+
+    [Header("Settings")]
+    [SerializeField] private float connectedWaypointRadius;
+    [SerializeField] private Collider2D[] hits = new Collider2D[7];
+    [SerializeField] private List<Waypoint> connectedWaypoints = new List<Waypoint>();
+
+    [Header("Debug")]
+    [SerializeField] private bool drawGizmos;
+
+    private Waypoint connectedWaypoint;
```

Comment:
```suggestion
    [Header("Refrences")]
    [SerializeField] private Grid parentGrid;

    [Header("Settings")]
    [SerializeField] private bool isOccupied;
    [SerializeField] private float connectedWaypointRadius;
    [SerializeField] private Collider2D[] hits = new Collider2D[8];

    [Header("Debug")]
    [SerializeField] private bool drawGizmos;

    private Waypoint _connectedWaypoint;
    private List<Waypoint> _connectedWaypoints = new List<Waypoint>();
```

1. connectedWaypoint mist een _
2. hits list count hoort 8 te zijn
3. connectedWaypoints kan private worden gemaakt
4. Tweaks aan de headers